FlowCrypt/src/main/java/com/flowcrypt/email/jetpack/workmanager/RefreshClientConfigurationWorker.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * © 2016-present FlowCrypt a.s. Limitations apply. Contact human@flowcrypt.com
+ * Contributors: DenBond7
+ */
+
+package com.flowcrypt.email.jetpack.workmanager
+
+import android.content.Context
+import androidx.work.Constraints
+import androidx.work.ExistingWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequestBuilder
+import androidx.work.WorkManager
+import androidx.work.WorkerParameters
+import com.flowcrypt.email.api.email.EmailUtil
+import com.flowcrypt.email.api.retrofit.FlowcryptApiRepository
+import com.flowcrypt.email.api.retrofit.request.model.LoginModel
+import com.flowcrypt.email.api.retrofit.response.api.ClientConfigurationResponse
+import com.flowcrypt.email.api.retrofit.response.api.DomainOrgRulesResponse
+import com.flowcrypt.email.api.retrofit.response.base.Result.Status
+import com.flowcrypt.email.util.GeneralUtil
+import com.flowcrypt.email.util.LogsUtil
+
+/**
+ * @author Denis Bondarenko
+ *         Date: 11/24/21
+ *         Time: 4:31 PM
+ *         E-mail: DenBond7@gmail.com
+ */
+class RefreshClientConfigurationWorker(context: Context, params: WorkerParameters) :
+  BaseWorker(context, params) {
+
+  override suspend fun doWork(): Result {
+    LogsUtil.d(TAG, "doWork")
+    val repository = FlowcryptApiRepository()
+    val publicEmailDomains = EmailUtil.getPublicEmailDomains()
+
+    for (account in roomDatabase.accountDao().getAccounts()) {
+      val domain = EmailUtil.getDomain(account.email)
+      if (domain in publicEmailDomains) {
+        continue
+      }
+
+      val fesUrl = GeneralUtil.generateFesUrl(domain)
+      val uuid = account.uuid ?: continue
+      try {
+        val result = repository.getDomainOrgRules(
+          context = applicationContext,
+          loginModel = LoginModel(account.email, uuid),
+          fesUrl = fesUrl
+        )
+
+        if (result.status == Status.SUCCESS) {
+          val fetchedOrgRules = (result.data as? DomainOrgRulesResponse)?.orgRules
+            ?: (result.data as? ClientConfigurationResponse)?.orgRules
+          fetchedOrgRules?.let { orgRules ->
+            roomDatabase.accountDao().updateSuspend(account.copy(clientConfiguration = orgRules))
+          }
+        }
+      } catch (e: Exception) {
+        e.printStackTrace()
+      }
+    }
+
+    LogsUtil.d(TAG, "work was finished")
+    return Result.success()
```

Comment:
@tomholub Please look at these lines. It's the main point of added functionality.